src/UI/Seller/src/app/buyers/components/buyers/buyer-edit/buyer-edit.component.ts

```diff
@@ -36,30 +41,63 @@ export class BuyerEditComponent {
 
   constructor(
     private buyerService: BuyerService,
+    private ocImpersonationService: OcImpersonationConfigService,
     private router: Router,
     private buyerTempService: BuyerTempService,
     private appAuthService: AppAuthService
   ) {}
 
   updateResourceFromEvent(event: any, field: string): void {
-    const value =
+    let resourceUpdate: ResourceFormUpdate;
+    if(field==="ImpersonatingEnabled") {
+      resourceUpdate= {
+        field: 'ImpersonationConfig',
+        value: this.showImpersonation ? null : this._superBuyerStatic?.ImpersonationConfig,
+        form: this.resourceForm
+      }
+      this.showImpersonation = !this.showImpersonation
+    } else {
+      const value =
       field === 'Buyer.Active' ? event.target.checked : event.target.value
-    const resourceUpdate = { field, value }
+      resourceUpdate = { 
+        field, 
+        value,
+        form: this.resourceForm
+      }
+    }
     this._superBuyerEditable = this.buyerService.getUpdatedEditableResource<HSBuyerPriceMarkup>(
       resourceUpdate,
       this._superBuyerEditable
     )
     this.checkForChanges()
   }
 
-  createBuyerForm(superBuyer: HSBuyerPriceMarkup): void {
-    const buyer = superBuyer.Buyer
-    const markup = superBuyer.Markup
+  createBuyerForm(superBuyer: any): void {
+    const {Buyer, Markup, ImpersonationConfig} = superBuyer;
+    this.showImpersonation = ImpersonationConfig && ImpersonationConfig !== null
+
     this.resourceForm = new FormGroup({
-      Name: new FormControl(buyer.Name, Validators.required),
-      Active: new FormControl(buyer.Active),
-      Markup: new FormControl(markup.Percent),
-      ChiliPublishFolder: new FormControl(buyer.xp.ChiliPublishFolder),
+      Name: new FormControl(Buyer.Name, Validators.required),
+      Active: new FormControl(Buyer.Active),
+      Markup: new FormControl(Markup.Percent),
+      ChiliPublishFolder: new FormControl(Buyer.xp.ChiliPublishFolder),
+      ImpersonatingEnabled: new FormControl(this.showImpersonation),
+      URL: new FormControl((Buyer.xp as any).URL),
```

Comment:
do we have a way to type the xp instead of using any?

Reply:
I updated the HSBuyer model in the middleware project. When we update the sdk the URL value will be on the buyer.xp here and we won't need to do this.